src/lib/support/tests/TestZclString.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements a unit test suite for ZclString
+ *
+ */
+
+#include <inttypes.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <nlunit-test.h>
+#include <support/CHIPMem.h>
+#include <support/CodeUtils.h>
+#include <support/UnitTestRegistration.h>
+
+#include <lib/support/ZclString.h>
+
+using namespace chip;
+using namespace chip::Logging;
+using namespace chip::Platform;
+
+static void TestZclStringLessThanMaximumSize(nlTestSuite * inSuite, void * inContext)
+{
+    uint8_t bufferMemory[256];
+    // Init zcl string
+    MutableByteSpan zclString(bufferMemory);
+    // Init c string
+    char * cString64 = static_cast<char *>(MemoryCalloc(1024, true));
+    // Add random chars to c string
+    for (size_t i = 0; i < 64; i++)
+        cString64[i] = 'A';
+
+    CHIP_ERROR err = MakeZclCharString(zclString, cString64);
+
+    // Memmory assert
+    NL_TEST_ASSERT(inSuite, cString64 != nullptr);
+    // ZCL String length assert
+    NL_TEST_ASSERT(inSuite, zclString.data()[0] == 64);
+    // ZCL String data is not null
+    NL_TEST_ASSERT(inSuite, zclString.data() != NULL);
+    // ZCL String data does not hold empty string
+    NL_TEST_ASSERT(inSuite, zclString.data()[0] != '\0');
+    // ZCL String is created with no errors
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
```

Comment:
This assert should be the first one....

Reply:
Moved null check and error check at the assertion top.